meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/util/ZkMetaClientUtil.java

```diff
@@ -0,0 +1,201 @@
+import org.apache.zookeeper.Op;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.data.ACL;
+import org.apache.zookeeper.server.EphemeralType;
+
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.function.Function;
+
+public class ZkMetaClientUtil {
+  //Default ACL value until metaClient Op has ACL of its own.
+  private static final List<ACL> DEFAULT_ACL = Collections.unmodifiableList(ZooDefs.Ids.OPEN_ACL_UNSAFE);
+
+  private ZkMetaClientUtil(){
+  }
+
+  /**
+   * Helper function for transactionOp. Converts MetaClient Op's into Zk Ops to execute
+   * zk transactional support.
+   * @param ops
+   * @return
+   */
+  public static List<Op> metaClientOpToZk(Iterable<org.apache.helix.metaclient.api.Op> ops) {
+    getOpMap().put(org.apache.helix.metaclient.api.Op.Type.CREATE, op -> {
+      try {
+        CreateMode mode = convertMetaClientMode(((org.apache.helix.metaclient.api.Op.Create) op).getEntryMode());
+        return Op.create(op.getPath(), ((org.apache.helix.metaclient.api.Op.Create) op).getData(), DEFAULT_ACL, mode);
+      } catch (KeeperException e) {
+        throw translateZkExceptionToMetaclientException(ZkException.create(e));
+      }
+    });
+
+    getOpMap().put(org.apache.helix.metaclient.api.Op.Type.DELETE,
+        op -> Op.delete(op.getPath(), ((org.apache.helix.metaclient.api.Op.Delete) op).getVersion()));
+
+    getOpMap().put(org.apache.helix.metaclient.api.Op.Type.SET,
+        op -> Op.setData(op.getPath(),
+            ((org.apache.helix.metaclient.api.Op.Set) op).getData(), ((org.apache.helix.metaclient.api.Op.Set) op).getVersion()));
+
+    getOpMap().put(org.apache.helix.metaclient.api.Op.Type.CHECK,
+        op -> Op.check(op.getPath(), ((org.apache.helix.metaclient.api.Op.Check) op).getVersion()));
+
+    List<Op> zkOps = new ArrayList<>();
+    for (org.apache.helix.metaclient.api.Op op : ops) {
+      Function<org.apache.helix.metaclient.api.Op, Op> function = getOpMap().get(op.getType());
+      if (function != null) {
+        zkOps.add(function.apply(op));
+      } else {
+        throw new IllegalArgumentException("Op type " + op.getType().name() + "is not supported.");
+      }
+    }
+    return zkOps;
+  }
+
+  private static final class OpmapHolder {
+    static final Map<org.apache.helix.metaclient.api.Op.Type, Function<org.apache.helix.metaclient.api.Op, Op>> OPMAP =
+        new EnumMap<>(org.apache.helix.metaclient.api.Op.Type.class);
+  }
+
+  private static Map<org.apache.helix.metaclient.api.Op.Type, Function<org.apache.helix.metaclient.api.Op, Op>> getOpMap() {
+    return OpmapHolder.OPMAP;
+  }
+
+  private static CreateMode convertMetaClientMode(MetaClientInterface.EntryMode entryMode) throws KeeperException {
+    if (entryMode == MetaClientInterface.EntryMode.PERSISTENT) {
+      return CreateMode.fromFlag(0);
```

Comment:
Should we use something verbose or informative than using hard coded values like 0, 1 or 4?

Reply:
+1 to Rahul's comment, I think we can just clearly type to return CreateMode.PERSISTENT or CreateMode.EPHEMERAL etc.

Reply:
Yes that makes more sense as the flag parameter provides no value. Thanks for the suggestions!